### ISSUE TITLE
can you solve my some wrong?

### ISSUE THREAD

**Issue by rw1995** [HiLab-git/DTC#9]
hi， author. The report is related with semi-supervision, but I can't find anything with unlabeled data.
I only see labeled data for training in Dataloader. In the code, how do you enter an unlabeled data into the network?
please give me some help. Thx.

**Comment by Luoxd1996** [comment]
Hi,
You can read the data loader carefully, and you also need to learn what is "batch sampler" in PyTorch. I recommend you read the official implementation of Mean Teacher.
Best,
Xiangde.

**Comment by rw1995** [comment]
你好，Luo。我仔细看了下代码中的相关逻辑，我查看了里面的代码。我查看了DataLoader 和 batchsampler 的补充。
我在其中主要产生的疑惑是这个样子的，望您能给出对我的解惑：
1. db_train是将80个训练数据都加载了进去，其中每个训练数据有存在{image,label}2个数据
2.  batch_sampler是以一种2+2的形式对labeled 和 unlabeled数据的索引进行排序， 然后在trainloader中 通过对应索引形式，将数据以一个batch一个batch的形式传递进去的。

但是， unlabeled数据从哪儿体现出来的呢？80个数据不是全部都有labeld的吗？

**Comment by rw1995** [comment]
@Luoxd1996  希望得到您的回复

**Comment by Luoxd1996** [comment]
Hi, 
Yes, all images have labels, but just used the first 16 images' annotation for training, and others are seen as unlabeled images. In addition, you can use two data loaders to load labeled images and unlabeled images separately, it is easily to implement. That's all!
Best,
Xiangde.

**Comment by rw1995** [comment]
> Hi,
> Yes, all images have labels, but just used the first 16 images' annotation for training, and others are seen as unlabeled images. In addition, you can use two data loaders to load labeled images and unlabeled images separately, it is easily to implement. That's all!
> Best,
> Xiangde.

嗨，Luo。我明白了你在代码中的意思。嘿嘿，第一次接触半监督学习，看的太急了，感谢你对我的回复，ღ( ´･ᴗ･` )比心

**Comment by rw1995** [comment]
@Luoxd1996  你好，Luo，我做了2个实验， 一个是在LA数据集上的强监督和你的半监督， 
强监督学习采用数据 16个，半监督学习采用数据16+64个， 网络和网络损失是一致的，采用你的代码中的结构。
但是我这儿实验结果是强监督学习要略好于半监督学习，这让我感到不可思议
实验结果如下：
'''python
[0.87793104 0.78924421 8.70535768 2.05542818]  强监督
[ 0.87942055  0.78776435 11.1678824   2.51834157]  半监督
dice = metric.binary.dc(pred, gt)
jc = metric.binary.jc(pred, gt)
hd = metric.binary.hd95(pred, gt)
asd = metric.binary.asd(pred, gt)
'''
请问能否解答这个问题

**Comment by Luoxd1996** [comment]
Hi,
Amazing results. But I don't know why, so I can not help you. In my experiments, our results are very similar with SASSNet, UAMT and many recent works reported results. Maybe you can try these works.
Best,
Xiangde.
